rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/resource/LRAParticipant.java

```diff
@@ -0,0 +1,168 @@
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian.resource;
+
+import io.narayana.lra.client.NarayanaLRAClient;
+import org.eclipse.microprofile.lra.annotation.Compensate;
+import org.eclipse.microprofile.lra.annotation.LRAStatus;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@ApplicationScoped
+@Path(LRAParticipant.RESOURCE_PATH)
+public class LRAParticipant {
+    public static final String RESOURCE_PATH = "participant2";
+
+    public static final String CREATE_OR_CONTINUE_LRA = "start-lra";
+    public static final String CREATE_OR_CONTINUE_LRA2 = "start-end-lra";
+    public static final String END_EXISTING_LRA = "end-lra";
+    public static final String CONTINUE_LRA = "continue-lra";
+    public static final String START_NEW_LRA = "start-lra2";
+
+    @Context
+    UriInfo uriInfo;
+
+    @Inject
+    private NarayanaLRAClient lraClient;
+
+    @GET
+    @Path(CREATE_OR_CONTINUE_LRA)
+    @LRA(value = LRA.Type.REQUIRED, end = false)
+    public Response beginLRAWithRemoteCalls(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lra1) {
+        assertEquals("lra1 should be active", LRAStatus.Active, lraClient.getStatus(lra1));
+
+        // start a new LRA
+        URI lra2 = remoteInvocation(lra1, START_NEW_LRA);
+        assertEquals("lra1 should still be active", LRAStatus.Active, lraClient.getStatus(lra1));
```

Comment:
It would be good to print the lra id in the assertion message. It's the same for other calls down here.

Reply:
Well the interesting point of the assertion is that it failed because the the LRA is in the wrong state. The point of unit tests is to detect regressions so that an engineer can be assigned to fix it (the actual value is not of interest). If you want this kind of information then we need to add trace logging into the test suite.

Reply:
Trace logging can be switch on.